common/options/options.go

```diff
@@ -383,7 +383,7 @@ func (auth *Auth) IsSet() bool {
 // but no password, and the authentication mechanism requires a password.
 func (auth *Auth) ShouldAskForPassword() bool {
 	return auth.Username != "" && auth.Password == "" &&
-		!(auth.Mechanism == "MONGODB-X509" || auth.Mechanism == "GSSAPI")
+		!(auth.Mechanism == "MONGODB-X509" || auth.Mechanism == "GSSAPI" || auth.Mechanism == "MONGODB-OIDC")
```

Comment:
Suggestion: rewrite this list of mechanisms as []slice and check slices.Contains.